RecoMET/METProducers/src/HcalNoiseInfoProducer.cc

```diff
@@ -672,137 +649,180 @@ HcalNoiseInfoProducer::filldigis(edm::Event& iEvent, const edm::EventSetup& iSet
-             // as the end of the data for this fiber
-             break;
-           }
-           else {
-             // if we don't have a match, then decrement the 
-             // starting TS and check again
-             start_ts--;
-           }
-         }
-
-         // now make some sanity checks on the determined overlap index
-         if( start_ts == -1 ) {
-           // if we didn't find any match, use the capID only to compare
-           if( latest_cap_match < 0 ) {
-             //this shouldn't happen, in this case use all the data from the fiber
-             start_ts = nts;
-           }
-           else {
-             // its possible that the timing of the fibers
-             // is shifted such that they do not overlap
-             // and we just want to stitch the fibers
-             // together with no removal.
-             // In this case the capIDs will match at the
-             // N-4 spot (and the ADCs will not)
-             // if this is not the case, then we just take
-             // the value of latest match
-             if( latest_cap_match == int(nts - 4) ) {
-               start_ts = nts;
-             } else {
-               start_ts = latest_cap_match;
-             }
-           }
-         }
-
-         // now store as the matching index
-         matching_idx.push_back(start_ts);
-       }
-
-       // for the last fiber we always use all of the data
-       matching_idx.push_back(10);
-
-       // now loop over the time slices of each fiber and make the sum
-       int icombts = -1;
-       for( unsigned fidx = 0 ; fidx < nFibers; ++fidx ) {
-         for( unsigned its = 0; its < matching_idx[fidx]; ++its ) {
-           icombts++;
-
-           // apply integration limits
-           if( icombts < laserMonitorTSStart_ ) continue;
-           if( laserMonitorTSEnd_ > 0 && icombts > laserMonitorTSEnd_ ) continue;
-
-           int adc = lasmon_adcs[fidx][its];
-
-           if( adc >= 0 ) { // skip invalid data
-             float fc = adc2fCHF[adc];
-             totalLasmonCharge += fc;
-           }
-
-         } 
-       }
-     } // if( fillLaserMonitor_ )
   } // if (hCalib.isValid()==true)
+  if( fillLaserMonitor_  && (hLasermon.isValid() == true) ) {
+
+    std::vector<std::vector<int> > lasmon_adcs(laserMonCBoxList_.size(), std::vector<int>());
+    std::vector<std::vector<int> > lasmon_capids(laserMonCBoxList_.size(), std::vector<int>());
+
+    unsigned max_nsamples = 0;
+    for(QIE10DigiCollection::const_iterator digi = hLasermon->begin(); digi != hLasermon->end(); digi++) {
+
+      QIE10DataFrame df = static_cast<QIE10DataFrame>(*digi);
+
+      HcalCalibDetId calibId( digi->id() );
+
+      // Fill the lasermonitor channels
+      int cboxch  = calibId.cboxChannel( );
+      int iphi    = calibId.iphi();
+      int ieta    = calibId.ieta();
+
+      // only check channels having the requested cboxch
+      if (std::find( laserMonCBoxList_.begin(), laserMonCBoxList_.end(),
```

Comment:
Is this ```find``` needed? If ```cboxch``` is found, then the loop over the same collection is re-done below; if it is not found, the check is similar as the ```if( cboxch == laserMonCBoxList_[idx])``` which is right at the beginning of the ```for``` loop below

Reply:
yes, we discussed this the first time I committed this code and it didn't get removed.  I'll remove it.